authentication.js

```diff
@@ -3,25 +3,125 @@
 // ------------------------------------
 // Eve Online ESI
 // ------------------------------------
+const _ = require('lodash')
 
 const EveOnlineSsoStrategy = require('passport-eveonline-sso').Strategy;
 
 module.exports = {
   init(passport, conf) {
-    console.log(conf)
-    passport.use(conf.key,
-      new EveOnlineSsoStrategy({
-        clientID: conf.clientID,
-        clientSecret: conf.clientSecret,
-        callbackURL: conf.callbackURL,
-      }, function(accessToken, refreshToken, profile, cb) {
-        console.log(profile)
-        WIKI.models.users.processProfile(profile).then((user) => {
-          return cb(null, user) || true
-        }).catch((err) => {
-          return cb(err, null) || true
+    var client = new EveOnlineSsoStrategy({
+      clientID: conf.clientID,
+      clientSecret: conf.clientSecret,
+      callbackURL: conf.callbackURL,
+      passReqToCallback: true,
+      scope: "publicData esi-characters.read_titles.v1",
+    }, async (req, accessToken, refreshToken, profile, cb) => {
+      try {
+
+        const failTitles = conf.disallowedTitles.split("\n")
+        const allowedCorps = conf.allowedCorps.split("\n")
+        // get allowed corp IDs
+        var res = await fetch('https://esi.evetech.net/latest/universe/ids/?datasource=tranquility&language=en', {
```

Comment:
This feels like a code smell to keep re-using `res`, but I suppose the `await` prevents anything bad from happening.

Reply:
Let's correct before merging.  I'll think of some more creative names 😁

Reply:
Also, just FYI. `let` is the declaration you can use if you know the variable with be reassigned (as opposed to `const`)